addons/spreadsheet/static/src/pivot/plugins/pivot_ui_plugin.js

```diff
@@ -197,6 +197,52 @@ export default class PivotUIPlugin extends spreadsheet.UIPlugin {
         return undefined;
     }
 
+    /**
+     * Return the domain args of a pivot formula from a position.
+     * For all those formulas:
+     *
+     * =ODOO.PIVOT(1,"expected_revenue","stage_id",2,"city","Brussels")
+     * =ODOO.PIVOT.HEADER(1,"stage_id",2,"city","Brussels")
+     * =ODOO.PIVOT.HEADER(1,"stage_id",2,"city","Brussels","measure","expected_revenue")
+     *
+     * the result is the same: ["stage_id", 2, "city", "Brussels"]
+     *
+     * If the cell is the result of ODOO.PIVOT.TABLE, the result is the domain of the cell
+     * as if it was the individual pivot formula
+     *
+     * @param {{ col: number, row: number, sheetId: string }} position
+     * @returns {(string | number)[] | undefined}
+     */
+    getPivotDomainArgsFromPosition(position) {
+        const cell = this.getters.getCorrespondingFormulaCell(position);
+        if (!cell || !cell.isFormula || getNumberOfPivotFormulas(cell.content) === 0) {
+            return undefined;
+        }
+        const mainPosition = this.getters.getCellPosition(cell.id);
+        const { args, functionName } = getFirstPivotFunction(cell.content);
+        if (functionName === "ODOO.PIVOT.TABLE") {
+            const pivotId = this.getters.getPivotIdFromPosition(position);
+            const dataSource = this.getPivotDataSource(pivotId);
+            if (!dataSource.isReady()) {
+                return undefined;
+            }
+            const pivotCol = position.col - mainPosition.col;
+            const pivotRow = position.row - mainPosition.row;
+            const pivotCells = this.getPivotTableStructure(pivotId).pivotCells;
+            const pivotCell = pivotCells[pivotCol][pivotRow];
```

Comment:
This code does not  take the absence/presence of the headers in the formula. pivotCells will contain the headers no matter what but we sometimes have to ignore it.